src/classes/RD2_QueueableService.cls

```diff
@@ -79,6 +90,32 @@ public class RD2_QueueableService {
         }
     }
 
+    /**
+    * @description Remove already Recurring Donation Ids that are already queue up to be evaluate to prevent being queue
+    * up the second time. Skip the removal if the process is synchronize since there will not be a potential race condition.
+    * @return Set<Id> Recurring Donation Ids 
+    */
+    private static Set<Id> removeAlreadyEnqueuedRecurringDonations(Set<Id> rdIds) {
+        if (Limits.getQueueableJobs() >= Limits.getLimitQueueableJobs()) {
```

Comment:
I think having this function run when we're going to process in synchronous mode should be ok too. We check limits again on line 85 before deciding whether or not to enqueue. Is there a reason we'd want to process all of the rdIds synchronously, including ones that may already be enqueued?

Reply:
We want to include all of them because of the following scenario: 
When updating the Opp and RD at the same time and the opp happens to update before RD. The first process reaches the queue limit, the evaluation run in sync and before the second update start, which will have a outdated result. When we get to the second update and exclude the RD, the result will be outdated.